articles/dms/howto-mysql-to-azure-mysql-powershell.md

```diff
@@ -0,0 +1,657 @@
+
+Execute the following script in MySQL Workbench on the target database to extract the drop trigger script and add trigger script.
+
+```sql
+SELECT
+	SchemaName,
+    GROUP_CONCAT(DropQuery SEPARATOR ';\n') as DropQuery,
+    Concat('DELIMITER $$ \n\n', GROUP_CONCAT(AddQuery SEPARATOR '$$\n'), '\n\nDELIMITER ;') as AddQuery
+FROM
+(
+SELECT 
+	TRIGGER_SCHEMA as SchemaName,
+	Concat('DROP TRIGGER ', TRIGGER_NAME) as DropQuery,
+    Concat('CREATE TRIGGER ', TRIGGER_NAME, ' ', ACTION_TIMING, ' ', EVENT_MANIPULATION, 
+			'\nON ', EVENT_OBJECT_TABLE, '\n' , 'FOR EACH ', ACTION_ORIENTATION, ' ',
+            ACTION_STATEMENT) as AddQuery
+FROM  
+	INFORMATION_SCHEMA.TRIGGERS
+) AS Queries
+GROUP BY SchemaName
+```
+
+Run the generated drop trigger query (DropQuery column) in the result to drop triggers in the target database. The add trigger query can be saved, to be used post data migration completion.
+
+## Step 1: Log in to your Microsoft Azure subscription
+
+Use the [Connect-AzAccount PowerShell command](/powershell/module/az.accounts/connect-azaccount) to sign in to your Azure subscription by using PowerShell, as per the directions in the article [Log in with Azure PowerShell](/powershell/azure/authenticate-azureps).
+
+The following script sets the default subscription for PowerShell session post login and creates a helper logging function for formatted console logs.
+
+```powershell
+[string] $SubscriptionName = "mySubscription"
+$ErrorActionPreference = "Stop";
+
+Connect-AzAccount
+Set-AzContext -Subscription $SubscriptionName
+$global:currentSubscriptionId = (Get-AzContext).Subscription.Id;
+
+function LogMessage([string] $Message, [bool] $IsProcessing = $false) {
+    if ($IsProcessing) {
+        Write-Host "$(Get-Date -Format "yyyy-MM-dd HH:mm:ss"): $Message" -ForegroundColor Yellow
+    }
+    else {
+        Write-Host "$(Get-Date -Format "yyyy-MM-dd HH:mm:ss"): $Message" -ForegroundColor Green
+    }    
+}
+```
+
+## Step 2: Register the Microsoft.DataMigration resource provider
+
+Registration of the resource provider needs to be done on each Azure subscription only once. Without the registration, you will not be able to create an instance of **Azure Database Migration Service**.
+
+Register the resource provider by using the [Register-AzResourceProvider](/powershell/module/az.resources/register-azresourceprovider) command. The following script registers the resource provider required for **Azure Database Migration Service**
+
+```powershell
+Register-AzResourceProvider -ProviderNamespace Microsoft.DataMigration
+```
+
+## Step 3: Create a resource group
+
+An Azure resource group is a logical container into which Azure resources are deployed and managed. Create a resource group before you can create a virtual machine.
```

Comment:
This mentions creating a virtual machine, why is the creation of a virtual machine necessary?  Or do we mean before they can create the DMS resource.

Reply:
Great catch .. done
Its meant for DMS resources